user/puzzles/template.py

```diff
@@ -0,0 +1,28 @@
+pkgname = "puzzles"
+pkgver = "20240802"
+pkgrel = 0
+build_style = "cmake"
+hostmakedepends = [
+    "cmake",
+    "halibut",
+    "imagemagick",
+    "ninja",
+    "perl",
+    "pkgconf"
+]
+makedepends = ["gtk+3-devel"]
+depends = ["cmd:xdg-open!xdg-utils"]
+pkgdesc = "Collection of small programs which implement one-player puzzle games"
+maintainer = "Earldridge Jazzed Pineda <earldridgejazzedpineda@gmail.com>"
+license = "MIT"
+url = "https://www.chiark.greenend.org.uk/~sgtatham/puzzles"
+source = "https://www.chiark.greenend.org.uk/~sgtatham/puzzles/puzzles.tar.gz"
```

Comment:
`puzzles` is the upstream name, but `sgt-puzzles` is also acceptable. There are actually two version numbers for Simon Tatham's Portable Puzzle Collection:
* 4 numerical components, which not used in the Unix version: 0.0.13937.0
* A date followed by a git commit hash: 20240802.1c1899e
The latter isn't allowed in Chimera Linux, because alphabetic characters are not allowed in version numbers unless they are followed by an underscore (_), and even then only specific sequences are allowed (e.g. alpha, beta, git, rc).

Upstream only keeps the newest version of Simon Tatham's Portable Puzzle Collection, so the package should be updated as soon as possible. The way I approached this in a non-Linux package manager is by using a "vanity URL", but I'm not sure if that's applicable here. The version number isn't on the homepage, so it should be obtained by accessing "https://www.chiark.greenend.org.uk/~sgtatham/puzzles/puzzles.tar.gz" which redirects to the filename containing the version number.

Reply:
> Upstream only keeps the newest version of Simon Tatham's Portable Puzzle Collection, so the package should be updated as soon as possible.

is there a source that does keep old tarballs? if not, otherwise you can just use the url that redirects to (just checked and it seems to be https://www.chiark.greenend.org.uk/~sgtatham/puzzles/puzzles-20240802.1c1899e.tar.gz), it's fine to update the url on every version bump (we already do this for e.g. epson-inkjet-printer-escpr)

Reply:
That has been fixed.